tests/integration_tests/test_upgrade.py

```diff
@@ -93,6 +96,26 @@ def test_upgrade(session_cloud: IntegrationCloud):
         instance.install_new_cloud_init(source, take_snapshot=False)
         instance.execute('hostname something-else')
         _restart(instance)
+        assert instance.execute('cloud-init status --wait --long').ok
         _output_to_compare(instance, after_path, netcfg_path)
 
     log.info('Wrote upgrade test logs to %s and %s', before_path, after_path)
+
+
+@pytest.mark.ci
+@pytest.mark.ubuntu
+def test_upgrade_package(session_cloud: IntegrationCloud):
+    if get_validated_source(session_cloud) != CloudInitSource.DEB_PACKAGE:
+        not_run_message = 'Test only supports upgrading to build deb'
+        if os.environ.get('TRAVIS'):
+            # If this isn't running on CI, we should know
+            pytest.fail(not_run_message)
+        else:
+            pytest.skip(not_run_message)
+
+    launch_kwargs = {'image_id': session_cloud._get_initial_image()}
```

Comment:
Should we promote `_get_initial_image` to the public API?

(Side note, which I noticed when reading `_get_initial_image`s code: should we be caching that value so that we avoid getting different "initial" images if we call that before and after an image publication?)

Reply:
What do you mean by before and after an image publication?

Although, now that I look at it again, should I just make [released_image_id](https://github.com/canonical/cloud-init/blob/master/tests/integration_tests/clouds.py#L114) public and use that instead?

Reply:
> What do you mean by before and after an image publication?

We'll call `_get_initial_image` once when `session_cloud` is first instantiated (in the entire test run), and once again here: if a release image is released between those two calls, we'll run this test and the other tests against different images.

> Although, now that I look at it again, should I just make [released_image_id](https://github.com/canonical/cloud-init/blob/master/tests/integration_tests/clouds.py#L114) public and use that instead?

Aha, yeah, I guess we are caching the value already! :+1: